lib/animas/animasSimulator.js

```diff
@@ -119,13 +168,30 @@ exports.make = function(config){
       }
 
       if(currStatus !== null) {
-        if(currStatus.status === 'suspended' && event.rate === 0) {
-          //this basal is suspended
-          event.deliveryType = 'suspend';
+        if(currStatus.status === 'suspended') {
+          if (event.rate === 0) {
+            //this basal is suspended
+            event.deliveryType = 'suspend';
+          }
+          else {
+            // basal is non-zero, but status still suspended
+            // is this a suspend event we created from an alarm?
+            // If so, we should submit a joined suspend/resume event
+            if(currStatus.reason.suspended === 'automatic') {
+
+              currStatus.duration = currBasal.duration;
```

Comment:
I'm not sure where the logic is going wrong here, but testing this out locally with my `no_power` alarm and subsequent suspend from yesterday, the suspended basal and the suspended deviceEvent are both getting a duration of 360000 - maybe somehow based on the scheduled basal? Not sure, but in this case my suspend only lasted from 22:56 to 23:02 so incorrect.